components/gantry/multiaxis/multiaxis.go

```diff
@@ -115,18 +116,28 @@ func (g *multiAxis) MoveToPosition(
 		return errors.Errorf("need position inputs for %v-axis gantry, have %v positions", len(g.subAxes), len(positions))
 	}
 
-	idx := 0
+	if len(positions) != len(g.lengthsMm) {
+		return errors.Errorf(
+			"number of input positions %v does not match total gantry axes length %v",
+			len(positions), len(g.lengthsMm),
+		)
+	}
+
+	jdx := 0
 	for _, subAx := range g.subAxes {
+
```

Comment:
I'd prefer never to start a block of code with a blank line. The change in indentation should be enough to indicate that something new is happening here.